Detectors/TPC/reconstruction/src/HwClusterer.cxx

```diff
@@ -433,18 +433,36 @@ void HwClusterer::ProcessTimeBins(int iTimeBinMin, int iTimeBinMax, MCLabelConta
   /*
    * collect clusters from individual cluster finder
    */
+
+  // map to count unique MC labels
+  std::map<MCCompLabel,int> labelCount;
+
+  // multiset to sort labels according to occurrence
+  auto mcComp = [](const std::pair<MCCompLabel, int>& a, const std::pair<MCCompLabel, int>& b) { return a.second > b.second;};
+  std::multiset<std::pair<MCCompLabel,int>,decltype(mcComp)> labelSort(mcComp);
+
+  // for each CRU
   for (unsigned cru = 0; cru < mClusterStorage.size(); ++cru) {
     std::vector<Cluster>* clustersFromCRU = &mClusterStorage[cru];
     std::vector<std::vector<std::pair<int,int>>>* labelsFromCRU = &mClusterDigitIndexStorage[cru];
 
+    // for each found cluster
     for(unsigned c = 0; c < clustersFromCRU->size(); ++c) {
       const auto clusterPos = mClusterArray->size();
       mClusterArray->emplace_back(clustersFromCRU->at(c));
+      labelCount.clear();
+      labelSort.clear();
+
+      // for each used digit
       for (auto &digitIndex : labelsFromCRU->at(c)) {
         if (digitIndex.first < 0) continue;
-        for (auto &l : mLastMcDigitTruth[digitIndex.second]->getLabels(digitIndex.first))
-          mClusterMcLabelArray->addElement(clusterPos,l);
+        for (auto &l : mLastMcDigitTruth[digitIndex.second]->getLabels(digitIndex.first)) {
+          try { labelCount.at(l) = labelCount.at(l)+1;}         // increment counter
```

Comment:
@KlewinS it will be much faster to use ``labelCount[l]++`` instead of try/catch

Reply:
indeed, using labelCount[l]++ faster

Reply:
i'd say "compacter" - nothing about std::map is fast :)

Reply:
speaking of which: since you use it in a tight loop - did you profile the code? maybe using a more performant container would help if you have a hotspot here - in many cases a simple sorted vector (or array) performs better with a binary lookup.

Reply:
@mkrzewic : don't exclude that sorted array will be faster, but the labelCount.at(l)++ is indeed faster than try/catch: e.g. ``for (int j=10;j--;) {labelCount[gRandom->Integer(10)]++; }`` is faster that the try/catch equivalent by factor >30.

Reply:
I don't think, this will be the bottleneck and it's done only once per event. But I will keep in mind that this could be improved if we have to speed up this part.

Reply:
@shahor02 factor 30 seems a bit much, but indeed, the "proper way" with the direct increment is faster by ~2X if we have mostly duplicated entries to ~7X is we always throw for all unique entries (with keys that are as large as here). (llvm 7.3, -O2, on my laptop).